azurerm/internal/services/policy/registration.go

```diff
@@ -29,9 +29,10 @@ func (r Registration) SupportedDataSources() map[string]*schema.Resource {
 // SupportedResources returns the supported Resources supported by this Service
 func (r Registration) SupportedResources() map[string]*schema.Resource {
 	return map[string]*schema.Resource{
-		"azurerm_policy_assignment":     resourceArmPolicyAssignment(),
-		"azurerm_policy_definition":     resourceArmPolicyDefinition(),
-		"azurerm_policy_set_definition": resourceArmPolicySetDefinition(),
-		"azurerm_policy_remediation":    resourceArmPolicyRemediation(),
+		"azurerm_policy_assignment":                               resourceArmPolicyAssignment(),
+		"azurerm_policy_definition":                               resourceArmPolicyDefinition(),
+		"azurerm_policy_set_definition":                           resourceArmPolicySetDefinition(),
+		"azurerm_policy_remediation":                              resourceArmPolicyRemediation(),
+		"azurerm_virtual_machine_configuration_policy_assignment": resourceVirtualMachineConfigurationPolicyAssignment(),
```

Comment:
Sorry if that was addressed in the old pr, i looked but did see it: there is assignment in the name yet the policy is inlined? does that mean this would make more sense as
```suggestion
		"azurerm_virtual_machine_configuration_policy": resourceVirtualMachineConfigurationPolicyAssignment(),
```

Reply:
My thought on this resource name is that we need to emphasize that this is an assignment like `azurerm_policy_assignment` - creation of this resource will make the policy enabled on the target, therefore I kept the word "assignment" in the resource name. With the assignment, this resource reads like a policy definition